File: cpp/161-170/Majority Element.cpp
class Solution {
public:
    int majorityElement(vector<int> &num) {
        stack<int> s;
        for(int i = 0;i < num.size();i++)
        {
            if(s.empty())
                s.push(num[i]);
            else{
                int t = s.top();
                if(t != num[i])
                    s.pop();
                else
                    s.push(num[i]);
            }
        }
        return s.top();
    }
};

File: python/161-170/Majority Element.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def majorityElement(self, nums):
        stack = []
        for e in nums:
            if len(stack) == 0:
                stack.append(e)
            else:
                if stack[-1] == e:
                    stack.append(e)
                else:
                    del stack[-1]
        return stack[0]
